src/mscorlib/src/System/Environment.cs

```diff
@@ -1052,8 +1086,27 @@ public static IDictionary GetEnvironmentVariables()
                 root[0]++;
             }
             return result;
+#else // PLATFORM_UNIX
+            // Use reflection to access the unix implementation in System.IO.FileSystem.DriveInfo.dll
+            Type driveInfoType = Type.GetType(
+                "System.IO.DriveInfo, System.IO.FileSystem.DriveInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken,
```

Comment:
> If the implementation of Environment class lived in corefx repo, would you use reflection to implement this as well?

Probably, unless System.IO.DriveInfo moved into System.Runtime.dll, which is unlikely.  Assuming it didn't, then to avoid reflection we'd essentially need to recompile all of System.IO.FileSystem.DriveInfo.dll into Environment, which doesn't seem worthwhile.

> it maybe be a good idea to use the same reflection forwarder across all platforms.

You mean use it for Windows as well?